accession/accession.py

```diff
@@ -1041,6 +1032,25 @@ def prepare_mad_qc_metric(self, gs_file: GSFile) -> Dict[str, Any]:
         mad_qc["attachment"] = attachment
         return mad_qc
 
+    def make_mad_qc_metric(self, encode_file: EncodeFile, gs_file: GSFile) -> None:
+        """
+        Special logic is required to facilitate situation where the experiment is
+        unreplicated in biological replication sense, but contains two technical replicates.
+        In this situation from pipeline POV the experiment is replicated and thus madQC gets calculated.
+        """
+        if encode_file.has_qc("MadQualityMetric"):
+            return
+        elif self.experiment.get_number_of_biological_replicates() > 2:
+            return
+        elif (
+            self.experiment.get_number_of_biological_replicates() == 1
+            and self.experiment.get_number_of_technical_replicates() == 2
+        ) or self.experiment.get_number_of_biological_replicates() == 2:
+            mad_qc = self.prepare_mad_qc_metric(gs_file)
+            return self.queue_qc(mad_qc, encode_file, "mad-quality-metric", shared=True)
+        else:
+            return
```

Comment:
`self.experiment.get_number_of_biological_replicates()` is called three times. Would be good to consolidate:
```suggestion
        if encode_file.has_qc("MadQualityMetric"):
            return
           
        num_biological_replicates = self.experiment.get_number_of_biological_replicates()
        if num_biological_replicates > 2:
            return
        elif (
            num_biological_replicates == 1
            and self.experiment.get_number_of_technical_replicates() == 2
        ) or num_biological_replicates == 2:
            mad_qc = self.prepare_mad_qc_metric(gs_file)
            return self.queue_qc(mad_qc, encode_file, "mad-quality-metric", shared=True)
        else:
            return
```

Reply:
Yeah, considered this and decided that three dict accesses is better than extra variable :) I'm ok, with either way.